src/pkg/actions/actions.go

```diff
@@ -0,0 +1,300 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package actions contains functions for running commands and tasks
+package actions
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/defenseunicorns/zarf/src/pkg/utils/exec"
+	"github.com/defenseunicorns/zarf/src/pkg/utils/helpers"
+	"github.com/defenseunicorns/zarf/src/pkg/variables"
+)
+
+// Spinner includes various messaging functions that are used in actions
+// TODO: Make this a real interface for progress
+type Spinner interface {
+	Errorf(err error, format string, a ...any)
+	Successf(format string, a ...any)
+	Updatef(format string, a ...any)
+	Write(p []byte) (n int, err error)
```

Comment:
Are you thinking this is a spinner or progressbar so someone may want to write progress? Would you ever be able to measure how much of an action is really done though?

Reply:
Spinner, the write method on spinner works differently from progressbar in that it takes a stdOut/stdErr stream